sites/fast-website/src/app/components/navigation/navigation.ts

```diff
@@ -1,7 +1,35 @@
-import { FASTElement } from "@microsoft/fast-element";
+import { FASTElement, observable } from "@microsoft/fast-element";
 import { applyMixins, StartEnd } from "@microsoft/fast-foundation";
 
-export class Navigation extends FASTElement {}
+export class Navigation extends FASTElement {
+    @observable
+    public opened: boolean = false;
+
+    @observable
+    public debounce: boolean = true;
+
+    private mediaQueryList: MediaQueryList;
+
+    private mqlListener = (e: MediaQueryListEvent): void => {
+        this.debounce = true;
+
+        if (!e.matches) {
+            this.opened = false;
+        }
+    };
+
+    constructor() {
+        super();
+
+        this.mediaQueryList = window.matchMedia("screen and (max-width: 800px)");
+        this.mediaQueryList.addListener(this.mqlListener);
```

Comment:
Should the listener cb get fired immediately if the view is loaded and it matches the matchMedia listener? If so, you might want to invoke your cb immediately because I don't think the attached listener will get fired in that case.

Reply:
@nicholasrice Since `debounce` is true by default and the component is instantiated with that value, the class would be persistent until the `MediaQueryListEvent` is triggered. Also I think triggering the nav button's click handler wouldn't be able to happen before the constructor finishes executing, but I'm not 100% on that.

I think the only way to guarantee that a `MediaQueryListEvent` happens after it's created is to force a window resize. But I'm confident that we don't need to do that here.